salesforce/bulkwrite.go

```diff
@@ -0,0 +1,299 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/spyzhov/ajson"
+)
+
+// BulkWriteParams defines how we are writing data to a SaaS API.
+type BulkWriteParams struct {
+	// The name of the object we are writing, e.g. "Account"
+	ObjectName string // required
+
+	// The external ID of the object instance we are updating. Provided in the case of UPDATE, but not CREATE.
+	ExternalId string // required
+
+	// The path to the CSV file we are writing
+	FilePath string // required
```

Comment:
Can we add one more param for the "mode" and have it be "upsert" or "insert"? Then `createJob` can create a different kind of job depending on if it's upsert or insert?